internal/home/control.go

```diff
@@ -320,6 +320,16 @@ func handleHTTPSRedirect(w http.ResponseWriter, r *http.Request) (ok bool) {
 		return false
 	}
 
+	// Add headers for HSTS
+	// This informs browsers that the site should only be accessed using HTTPS,
+	// and that any future attempts to access it using HTTP should automatically be
+	// converted to HTTPS.
+	//
+	// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+	if config.TLS.ForceHTTPS {
+		w.Header().Add("Strict-Transport-Security", "max-age=63072000; includeSubDomains")
```

Comment:
1.  From what I've read in the MDN article and in some recommendations, HSTS headers should only be set in HTTPS responses.  It looks like this code will set it regardless of whether or not the request is HTTPS?

2.  Let's limit max-age to 1 year for now.

Reply:
> 1. From what I've read in the MDN article and in some recommendations, HSTS headers should only be set in HTTPS responses.  It looks like this code will set it regardless of whether or not the request is HTTPS?
> 2. Let's limit max-age to 1 year for now.

@ainar-g , ForceHTTPS, validation is added, which could be enabled, only if https is enabled, and force http -> https redirection is done in AG

Will change it to 1 year

Reply:
@ssrahul96, consider the following case: if a client accesses AGH using plain HTTP, they are redirected, but the redirection response will contain the header, thus transferring it over plain HTTP.  This is the situation I'd like to avoid.

Reply:
this will occur for the very 1st api call, from next call onwards hsts would come into picture, anyhow will see, if i can handle it

Reply:
moved the hsts header block after `https` redirection, that should resolved this